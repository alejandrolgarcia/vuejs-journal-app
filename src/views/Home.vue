<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <div class="btn btn-primary">Primary</div>
  </div>
</template>
