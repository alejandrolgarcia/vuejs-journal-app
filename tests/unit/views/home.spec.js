import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Pruebas en el Home View', () => {

    test('debe de renderizar el componente correctamente', () => {
        const wrapper = shallowMount( Home )
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('al hacer clic en el boton debe redireccionar a no-entry', () => {
        
        // crear nuestro mock
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount( Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        // estimulo
        wrapper.find('button').trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry'})

    })

});
