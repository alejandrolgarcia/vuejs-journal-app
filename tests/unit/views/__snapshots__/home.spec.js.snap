// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home View debe de renderizar el componente correctamente 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-primary">Primary</button></div>`;
